PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/resources/META-INF/ruleConfigs.json

```diff
@@ -45,5 +45,16 @@
     "methods_to_check": "connectionString",
     "anti_pattern_message": "Connection String detected. Use DefaultAzureCredential for Azure service client authentication instead if the service client supports Token Credential (Entra ID Authentication).",
     "solution": "DefaultAzureCredential is recommended if the service client supports Token Credential (Entra ID Authentication). if not, then use Azure Key Credential / Connection Strings based authentication"
+  },
+  "KustoQueriesWithTimeIntervalInQueryStringCheck": {
+    "regex_patterns": {
+      "KQL_ANTI_PATTERN_AGO": ".*ago\\(",
+      "KQL_ANTI_PATTERN_DATETIME": ".*datetime\\s*\\(",
+      "KQL_ANTI_PATTERN_NOW": ".*now\\(",
+      "KQL_ANTI_PATTERN_START_OF_PERIOD": ".*startofday\\(\\)|.*startofmonth\\(\\)|.*startofyear\\(\\)",
+      "KQL_ANTI_PATTERN_BETWEEN": ".*between\\(datetime\\("
+    },
+    "anti_pattern_message": "KQL queries with time intervals in the query string detected.",
+    "solution": "Use the QueryTimeInterval parameter in the client method parameters to specify the time interval for the query"
```

Comment:
Is there any in-depth guidance we can link to for the solution? It'd be nice to have examples of better patterns vs the patterns we want to recommend against.

Reply:
This is the only link I found with a good description of what Srikanta's suggestion for the alternative was: [QueryTimeInterval Class](https://learn.microsoft.com/en-us/java/api/overview/azure/monitor-query-readme?view=azure-java-stable) -- So i tried to make the README a bit more descriptive

I didn't find a reference to QueryTimeInterval in the [Azure-kust-java repo](https://github.com/Azure/azure-kusto-java)